packages/core/src/indexing/postgres/__tests__/postgres-index-api.test.ts

```diff
@@ -59,25 +46,47 @@ describe('init', () => {
       const modelsToIndex = [StreamID.fromString(STREAM_ID_A)]
       const indexApi = new PostgresIndexApi(dbConnection, modelsToIndex, true)
       await indexApi.init()
-      const created = await listMidTables()
+      const created = await listMidTables(dbConnection)
+      const tableNames = modelsToIndex.map(asTableName)
+      expect(created).toEqual(tableNames)
+    })
+
+    test('create new table from scratch but fail table validation', async () => {
+      const INVALID_TABLE_STRUCTURE = {
+        stream_id: {
+          type: 'character varying',
+          maxLength: 254,
+          nullable: false,
+          defaultValue: null,
+        },
+      }
+
+      const modelsToIndex = [StreamID.fromString(STREAM_ID_A)]
+      const indexApi = new PostgresIndexApi(dbConnection, modelsToIndex, true)
+      await indexApi.init()
+      const created = await listMidTables(dbConnection)
       const tableNames = modelsToIndex.map(asTableName)
       expect(created).toEqual(tableNames)
+      // TODO: error check is not caught correctly... why jest... why u no workie
+      /*await expect(() => {
+        indexApi.verify(INVALID_TABLE_STRUCTURE)
+      }).toThrowError(/Schema verification failed for index/)*/
```

Comment:
```suggestion
      }).rejects.toThrow(/Schema verification failed for index/)*/
```

Reply:
Does that change things?

Reply:
No, tried that unfortunately already. Also a nested version I found in a different test file. But somehow its just not being caught.

Reply:
Welp, I stand corrected. That did the trick - I was sure I used this before though.